gobblin-data-management/src/main/java/gobblin/data/management/copy/writer/TarArchiveInputStreamDataWriter.java

```diff
@@ -62,20 +63,27 @@ public void write(FileAwareInputStream fileAwareInputStream) throws IOException
 
     TarArchiveInputStream tarIn = new TarArchiveInputStream(fileAwareInputStream.getInputStream());
     TarArchiveEntry tarEntry;
-    boolean firstEntrySeen = false;
+
+    Path fileDestinationPath = fileAwareInputStream.getFile().getDestination();
+
+    // root entry in the tar compressed file
+    String tarEntryRootName = null;
 
     try {
       while ((tarEntry = tarIn.getNextTarEntry()) != null) {
 
-        Path tarEntryPath =
-            PathUtils.withoutLeadingSeparator(new Path(fileAwareInputStream.getFile().getDestination().getParent(),
-                tarEntry.getName()));
+        if (tarEntryRootName == null) {
```

Comment:
See next line 76. We set the tarEntryRootName. This is only true for the first tarEntry